compiler/qsc/src/interpret/stateful.rs

```diff
@@ -264,7 +264,7 @@ impl Interpreter {
         };
 
         eval_expr(
-            &mut self.state,
+            &mut State::new(self.source_package),
```

Comment:
I'm curious about why a new state is created here rather than `self.state`.

Reply:
If there is an exception, we abort execution but do not clear the "tape" used to track execution. This means it can leave leftover expressions in the state that will cause failures on subsequent runs. We fixed a variation of this bug for Python at one point, but I missed it when implementing this variation and incorrectly reused the state from `self`. This works fine as long as execution completes, since that means the tape in the state gets fully evaluated.

Reply:
"Exception" is the wrong word... I mean a Q# runtime failure, like a divide by zero or an explicit `fail`, or in the sample case, a qubit released in a non-zero state.

Reply:
I see, thank you for the explanation!